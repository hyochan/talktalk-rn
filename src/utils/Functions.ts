import { Animated } from 'react-native';

export const animateRotateLoop = (spinValue, duration?: number) => {
  if (!duration) {
    duration = 2000;
  }
  /**
   * Animation for buffering image.
   * First set up animation
   * Second interpolate beginning and end values (in this case 0 and 1)
   */
  Animated.loop(
    Animated.sequence([
      Animated.timing(spinValue, {
        toValue: 1,
        duration,
        delay: 0,
      }),
      Animated.timing(spinValue, {
        toValue: 0,
        duration,
      }),
    ]),
    {
      iterations: 4,
    },
  ).start();
};
