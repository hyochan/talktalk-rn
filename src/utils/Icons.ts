export const IC_MASK = require('../../assets/icons/mask.png');
export const IC_ICON = require('../../assets/icons/icon.png');
export const IC_BACK = require('../../assets/icons/btnBack.png');
export const IC_ADD = require('../../assets/icons/btnAdd.png');
export const IC_PIC = require('../../assets/icons/pictoGallery.png');
export const IC_SEARCH = require('../../assets/icons/picSearch.png');
export const IC_SMILE = require('../../assets/icons/icSmile.png');
