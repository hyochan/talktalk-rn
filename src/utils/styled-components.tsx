import * as styledComponents from "styled-components/native";
import ThemeInterface from "./theme";

const {
  default: styled,
  css,
  ThemeProvider
} = styledComponents as styledComponents.ReactNativeThemedStyledComponentsModule<typeof ThemeInterface>;

export { css, ThemeProvider };
export default styled;