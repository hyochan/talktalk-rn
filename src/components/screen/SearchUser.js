// @flow
import React, { Component, useState, useContext } from 'react';
import {
  StyleSheet,
  TouchableOpacity,
  Image,
  Text,
  View,
  TextInput,
  Animated,
  FlatList,
  SafeAreaView,
} from 'react-native';
import { NavigationScreenProps } from 'react-navigation';
import { getString } from '../../../STRINGS';
import { User as Friend } from '../../models/User';
import { IC_BACK, IC_SEARCH, IC_ICON } from '../../utils/Icons';

import type {
  ____ViewStyleProp_Internal as ViewStyle,
  ____TextStyleProp_Internal as TextStyle,
  ____ImageStyleProp_Internal as ImageStyle,
} from 'react-native/Libraries/StyleSheet/StyleSheetTypes';

import { ratio, colors } from '../../utils/Styles';
import { ProfileModalContext } from '../../providers/ProfileModalProvider';
import EmptyListItem from '../shared/EmptyListItem';
import UserListItem from '../shared/UserListItem';

import styled from 'styled-components/native';

const userSampleData = [
  {
    uid: '1', displayName: 'test', photoURL: null, statusMsg: 'status', isOnline: '', friends: '', Chatrooms: '', created: '', updated: '',
  },
  {
    uid: '2', displayName: 'geoseong', photoURL: IC_BACK, statusMsg: 'healthy', isOnline: '', friends: '', Chatrooms: '', created: '', updated: '',
  },
  {
    uid: '3', displayName: 'hyochan', photoURL: IC_ICON, statusMsg: 'healthy', isOnline: '', friends: '', Chatrooms: '', created: '', updated: '',
  },
  {
    uid: '4', displayName: 'test', photoURL: null, statusMsg: 'status', isOnline: '', friends: '', Chatrooms: '', created: '', updated: '',
  },
  {
    uid: '5', displayName: 'geoseong', photoURL: IC_BACK, statusMsg: 'healthy', isOnline: '', friends: '', Chatrooms: '', created: '', updated: '',
  },
  {
    uid: '6', displayName: 'hyochan', photoURL: IC_ICON, statusMsg: 'healthy', isOnline: '', friends: '', Chatrooms: '', created: '', updated: '',
  },
  {
    uid: '7', displayName: 'test', photoURL: null, statusMsg: 'status', isOnline: '', friends: '', Chatrooms: '', created: '', updated: '',
  },
  {
    uid: '8', displayName: 'geoseong', photoURL: IC_BACK, statusMsg: 'healthy', isOnline: '', friends: '', Chatrooms: '', created: '', updated: '',
  },
  {
    uid: '9', displayName: 'hyochan', photoURL: IC_ICON, statusMsg: 'healthy', isOnline: '', friends: '', Chatrooms: '', created: '', updated: '',
  },
];
const AnimatedFlatList = Animated.createAnimatedComponent(FlatList);
const StyledSafeAreaView = styled.SafeAreaView`
  flex: 1;
`;
const StyledContainer = styled.View`
  flex: 1;
  background-color: transparent;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`;
const StyledSearchView = styled.View`
  width: 100%;
  height: 50;
  justify-content: center;
  overflow: hidden;
`;
const StyledAnimatedFlatList = styled(AnimatedFlatList)`
  width: 100%;
  height: 100%;
`;
const StyledTextInputWrapper = styled.View`
  width: 100%;
  height: 50;
  position: absolute;
  padding-horizontal: 20;
`;
const StyledTextInput = styled.TextInput`
  width: 100%;
  height: 30;
  top: 10;
  background-color: rgb(247,248,251);
  border-radius: 4;
  padding-left: 34;
  padding-right: 10;
`;
const StyledSearchImage = styled.Image`
  width: 16;
  height: 16;
  position: absolute;
  top: 18;
  left: 30;
`;

type Props = {
  navigation: NavigationScreenProps,
};
type State = {
  users: Friend[],
  searchedUsers: Friend[],
};

Screen.navigationOptions = {
  title: 'Search User',
};
function Screen(props: Props) {
  const profileModal = React.useContext(ProfileModalContext);
  const [searchedUsers, setSearchedUsers] = useState(userSampleData);
  const [users, setUsers] = useState(userSampleData);
  const scrollY = new Animated.Value(0);

  const renderItem = ({ item }: Friend) => {
    return (
      <UserListItem
        testID='userListItem'
        user={item}
        onPress={() => {
          if (profileModal && profileModal.dispatch) {
            profileModal.dispatch({
              type: 'show-modal',
              payload: { user: item, deleteMode: true },
            });
          }
        }}
      />
    );
  };
  const onTxtChanged = (txt: string) => {
    onSearch(txt);
    scrollY.setValue(0);
    Animated.timing(scrollY, {
      toValue: 100,
      duration: 500,
    }).start();
  };
  const onSearch = (txt: string) => {
    const searchedUser = (txt === '')
      ? searchedUsers
      : searchedUsers.filter((item) => item.displayName.includes(txt));
    setUsers(searchedUser);
  };
  const getContentContainerStyle = () => {
    return users.length === 0
      ? {
        flex: 1,
        alignItems: 'center',
        justifyContent: 'center',
      }
      : null;
  };
  return (
    <StyledSafeAreaView>
      <StyledContainer>
        <StyledSearchView>
          <StyledTextInputWrapper>
            <StyledTextInput
              testID="styledInput"
              onChangeText={onTxtChanged}
              underlineColorAndroid='transparent' // android fix
              autoCapitalize='none'
              autoCorrect={false}
              multiline={false}
              defaultValue={''}
            />
            <StyledSearchImage source={IC_SEARCH} />
          </StyledTextInputWrapper>
        </StyledSearchView>
        <StyledAnimatedFlatList
          testID='animatedFlatlist'
          style={{
            transform: [{
              translateY: scrollY.interpolate({
                inputRange: [0, 50, 100],
                outputRange: [0, 25, 0],
              })
            }]
          }}
          contentContainerStyle={getContentContainerStyle}
          keyExtractor={(item, index) => index.toString()}
          data={users}
          renderItem={renderItem}
          ListEmptyComponent={<EmptyListItem>{getString('NO_CONTENT')}</EmptyListItem>}
        />
      </StyledContainer>
    </StyledSafeAreaView>
  );
}

export default Screen;
