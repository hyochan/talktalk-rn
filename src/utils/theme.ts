import { Dimensions, Platform } from 'react-native';
import { getStatusBarHeight } from 'react-native-status-bar-height';
import { DefaultTheme } from 'styled-components';

const { width, height } = Dimensions.get('window');
let calRatio = width <= height ? 16 * (width / height) : 16 * (height / width);
if (width <= height) {
  if (calRatio < 9) {
    calRatio = width / 9;
  } else {
    calRatio = height / 18;
  }
} else {
  if (calRatio < 9) {
    calRatio = height / 9;
  } else {
    calRatio = width / 18;
  }
}

const theme: DefaultTheme = {
  screenWidth: width,
  screenHeight: height,
  ratio: calRatio / (360 / 9),
  statusBarHeight: getStatusBarHeight(true),
  colors: {
    background: '#e3e3e3',
    dodgerBlue: 'rgb(58,139,255)',
    dusk: 'rgb(65,77,107)',
    blueyGray: 'rgb(134,154,183)',
    cloudyBlue: 'rgb(175,194,219)',
    greenishCyan: 'rgb(80,227,194)',
    paleGray: 'rgb(233,237,244)',
  }
};

export default theme;