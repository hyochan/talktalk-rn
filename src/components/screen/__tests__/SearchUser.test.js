import { Animated } from 'react-native';
import * as React from 'react';
import SearchUser from '../SearchUser';
import { IC_BACK } from '../../../utils/Icons';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { RenderAPI, render, shallow, fireEvent, waitForElement } from 'react-native-testing-library';

const props = {
  navigation: {
    navigate: jest.fn(),
  },
  onTxtChanged: jest.fn(),
  onSearch: jest.fn(),
};
const component: React.Element<any> = <SearchUser {...props}/>;
// const component: React.Element<any> = <SearchUser {...props}/>;

describe('[SearchUser] rendering test', () => {
  it('renders as expected', () => {
    const json = renderer.create(component).toJSON();
    expect(json).toMatchSnapshot();
  });
});

// jest.useFakeTimers(); // related to Animated Component
describe('[serachUser] interaction', () => {
  let testingLib: RenderAPI
  let txtInputInst: renderer.ReactTestInstance
  let animatedFlatlistInst: renderer.ReactTestInstance
  const inputData = {
    uid: '2',
    displayName: 'geoseong',
    photoURL: IC_BACK,
    statusMsg: 'healthy',
    isOnline: '',
    friends: '',
    Chatrooms: '',
    created: '',
    updated: '',
  };
  
  beforeAll(() => {
    // jest.useFakeTimers(); // related to Animated Component
    testingLib = render(component);
    txtInputInst = testingLib.getByTestId('txtInput')
    animatedFlatlistInst = testingLib.getByTestId('animatedFlatlist')
  });
  it('when friend name typed in TextInput: (onTxtChanged -> onSearch) and (renderItem)', () => {
    // setTimeout called - 0
    fireEvent.changeText(txtInputInst, inputData.displayName)
    // setTimeout called - 2
    expect(animatedFlatlistInst.props.data[0]).toEqual(inputData);
    
    expect(animatedFlatlistInst.props.contentContainerStyle()).toEqual(null);
    fireEvent.changeText(txtInputInst, 'noresult!')
    // setTimeout called - 3
    expect(animatedFlatlistInst.props.data.length).toEqual(0);
    expect(animatedFlatlistInst.props.contentContainerStyle()).toEqual({
      flex: 1,
      alignItems: 'center',
      justifyContent: 'center',
    });
    
    fireEvent.changeText(txtInputInst, '')
    // setTimeout called - 4
    expect(animatedFlatlistInst.props.data.length).toEqual(9);
  });
  it('see Animation is working well', () => {
    jest.useFakeTimers(); // related to timer
    let scrollY = animatedFlatlistInst.props.testObj.scrollY;
    let afterScrollY = new Animated.Value(0)
    Animated.timing(afterScrollY, {
      toValue: 100,
      duration: 500,
    }).start();
    // setTimeout called - 5
    setTimeout(() => {
      console.log('after scrollY', afterScrollY)
      expect(scrollY).toEqual(afterScrollY);  // doesn't work
    }, 600);
    // setTimeout called - 6
  });
  
  it('when profile modal clicked -> should call showProfileModal', () => {
    fireEvent.changeText(txtInputInst, inputData.displayName);
    let userListItemInst: renderer.ReactTestInstance = testingLib.getByTestId('userListItem0');
    fireEvent.press(userListItemInst);
    let { profileModal } = userListItemInst.props.testObj;  // try to detect cotnext, but undefined
  });
});