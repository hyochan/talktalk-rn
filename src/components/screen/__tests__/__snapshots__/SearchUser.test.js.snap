// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[SearchUser] rendering test renders as expected 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flexBasis": 0,
          "flexDirection": "column",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "height": 50,
            "justifyContent": "center",
            "overflow": "hidden",
            "width": "100%",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "height": 50,
              "paddingHorizontal": 20,
              "position": "absolute",
              "width": "100%",
            },
          ]
        }
      >
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          autoCorrect={false}
          defaultValue=""
          multiline={false}
          onChangeText={[Function]}
          rejectResponderTermination={true}
          style={
            Array [
              Object {
                "backgroundColor": "rgb(247,248,251)",
                "borderRadius": 4,
                "height": 30,
                "paddingLeft": 34,
                "paddingRight": 10,
                "top": 10,
                "width": "100%",
              },
            ]
          }
          testID="txtInput"
          underlineColorAndroid="transparent"
        />
        <Image
          source={
            Object {
              "testUri": "../../../assets/icons/picSearch.png",
            }
          }
          style={
            Array [
              Object {
                "height": 16,
                "left": 30,
                "position": "absolute",
                "top": 18,
                "width": 16,
              },
            ]
          }
        />
      </View>
    </View>
    <RCTScrollView
      ListEmptyComponent={
        <Shared
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "stretch",
              "flex": 1,
              "justifyContent": "center",
              "width": "100%",
            }
          }
        >
          No content
        </Shared>
      }
      contentContainerStyle={[Function]}
      data={
        Array [
          Object {
            "Chatrooms": "",
            "created": "",
            "displayName": "test",
            "friends": "",
            "isOnline": "",
            "photoURL": null,
            "statusMsg": "status",
            "uid": "1",
            "updated": "",
          },
          Object {
            "Chatrooms": "",
            "created": "",
            "displayName": "geoseong",
            "friends": "",
            "isOnline": "",
            "photoURL": Object {
              "testUri": "../../../assets/icons/btnBack.png",
            },
            "statusMsg": "healthy",
            "uid": "2",
            "updated": "",
          },
          Object {
            "Chatrooms": "",
            "created": "",
            "displayName": "hyochan",
            "friends": "",
            "isOnline": "",
            "photoURL": Object {
              "testUri": "../../../assets/icons/icon.png",
            },
            "statusMsg": "healthy",
            "uid": "3",
            "updated": "",
          },
          Object {
            "Chatrooms": "",
            "created": "",
            "displayName": "test",
            "friends": "",
            "isOnline": "",
            "photoURL": null,
            "statusMsg": "status",
            "uid": "4",
            "updated": "",
          },
          Object {
            "Chatrooms": "",
            "created": "",
            "displayName": "geoseong",
            "friends": "",
            "isOnline": "",
            "photoURL": Object {
              "testUri": "../../../assets/icons/btnBack.png",
            },
            "statusMsg": "healthy",
            "uid": "5",
            "updated": "",
          },
          Object {
            "Chatrooms": "",
            "created": "",
            "displayName": "hyochan",
            "friends": "",
            "isOnline": "",
            "photoURL": Object {
              "testUri": "../../../assets/icons/icon.png",
            },
            "statusMsg": "healthy",
            "uid": "6",
            "updated": "",
          },
          Object {
            "Chatrooms": "",
            "created": "",
            "displayName": "test",
            "friends": "",
            "isOnline": "",
            "photoURL": null,
            "statusMsg": "status",
            "uid": "7",
            "updated": "",
          },
          Object {
            "Chatrooms": "",
            "created": "",
            "displayName": "geoseong",
            "friends": "",
            "isOnline": "",
            "photoURL": Object {
              "testUri": "../../../assets/icons/btnBack.png",
            },
            "statusMsg": "healthy",
            "uid": "8",
            "updated": "",
          },
          Object {
            "Chatrooms": "",
            "created": "",
            "displayName": "hyochan",
            "friends": "",
            "isOnline": "",
            "photoURL": Object {
              "testUri": "../../../assets/icons/icon.png",
            },
            "statusMsg": "healthy",
            "uid": "9",
            "updated": "",
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      style={
        Object {
          "height": "100%",
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
          "width": "100%",
        }
      }
      testID="animatedFlatlist"
      testObj={
        Object {
          "scrollY": 0,
        }
      }
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "width": "100%",
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "borderBottomWidth": 1,
                    "borderColor": "rgb(245,245,245)",
                    "flexDirection": "row",
                    "height": 80,
                    "justifyContent": "flex-start",
                    "paddingHorizontal": 20,
                  }
                }
              >
                <Unknown
                  color="rgb(65,77,107)"
                  light={true}
                  name="meh"
                  size={40}
                />
                <Text
                  style={
                    Object {
                      "color": "rgb(65,77,107)",
                      "fontSize": 14,
                      "marginLeft": 12,
                      "width": 100,
                    }
                  }
                >
                  test
                </Text>
                <Text
                  style={
                    Object {
                      "borderColor": "rgb(233,237,244)",
                      "borderWidth": 0.3,
                      "color": "rgb(65,77,107)",
                      "fontSize": 12,
                      "maxWidth": 134.2,
                      "paddingHorizontal": 8,
                      "paddingVertical": 4,
                      "position": "absolute",
                      "right": 20,
                    }
                  }
                >
                  status
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "width": "100%",
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "borderBottomWidth": 1,
                    "borderColor": "rgb(245,245,245)",
                    "flexDirection": "row",
                    "height": 80,
                    "justifyContent": "flex-start",
                    "paddingHorizontal": 20,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "testUri": "../../../assets/icons/btnBack.png",
                    }
                  }
                  style={
                    Object {
                      "height": 40,
                      "width": 40,
                    }
                  }
                />
                <Text
                  style={
                    Object {
                      "color": "rgb(65,77,107)",
                      "fontSize": 14,
                      "marginLeft": 12,
                      "width": 100,
                    }
                  }
                >
                  geoseong
                </Text>
                <Text
                  style={
                    Object {
                      "borderColor": "rgb(233,237,244)",
                      "borderWidth": 0.3,
                      "color": "rgb(65,77,107)",
                      "fontSize": 12,
                      "maxWidth": 134.2,
                      "paddingHorizontal": 8,
                      "paddingVertical": 4,
                      "position": "absolute",
                      "right": 20,
                    }
                  }
                >
                  healthy
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "width": "100%",
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "borderBottomWidth": 1,
                    "borderColor": "rgb(245,245,245)",
                    "flexDirection": "row",
                    "height": 80,
                    "justifyContent": "flex-start",
                    "paddingHorizontal": 20,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "testUri": "../../../assets/icons/icon.png",
                    }
                  }
                  style={
                    Object {
                      "height": 40,
                      "width": 40,
                    }
                  }
                />
                <Text
                  style={
                    Object {
                      "color": "rgb(65,77,107)",
                      "fontSize": 14,
                      "marginLeft": 12,
                      "width": 100,
                    }
                  }
                >
                  hyochan
                </Text>
                <Text
                  style={
                    Object {
                      "borderColor": "rgb(233,237,244)",
                      "borderWidth": 0.3,
                      "color": "rgb(65,77,107)",
                      "fontSize": 12,
                      "maxWidth": 134.2,
                      "paddingHorizontal": 8,
                      "paddingVertical": 4,
                      "position": "absolute",
                      "right": 20,
                    }
                  }
                >
                  healthy
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "width": "100%",
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "borderBottomWidth": 1,
                    "borderColor": "rgb(245,245,245)",
                    "flexDirection": "row",
                    "height": 80,
                    "justifyContent": "flex-start",
                    "paddingHorizontal": 20,
                  }
                }
              >
                <Unknown
                  color="rgb(65,77,107)"
                  light={true}
                  name="meh"
                  size={40}
                />
                <Text
                  style={
                    Object {
                      "color": "rgb(65,77,107)",
                      "fontSize": 14,
                      "marginLeft": 12,
                      "width": 100,
                    }
                  }
                >
                  test
                </Text>
                <Text
                  style={
                    Object {
                      "borderColor": "rgb(233,237,244)",
                      "borderWidth": 0.3,
                      "color": "rgb(65,77,107)",
                      "fontSize": 12,
                      "maxWidth": 134.2,
                      "paddingHorizontal": 8,
                      "paddingVertical": 4,
                      "position": "absolute",
                      "right": 20,
                    }
                  }
                >
                  status
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "width": "100%",
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "borderBottomWidth": 1,
                    "borderColor": "rgb(245,245,245)",
                    "flexDirection": "row",
                    "height": 80,
                    "justifyContent": "flex-start",
                    "paddingHorizontal": 20,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "testUri": "../../../assets/icons/btnBack.png",
                    }
                  }
                  style={
                    Object {
                      "height": 40,
                      "width": 40,
                    }
                  }
                />
                <Text
                  style={
                    Object {
                      "color": "rgb(65,77,107)",
                      "fontSize": 14,
                      "marginLeft": 12,
                      "width": 100,
                    }
                  }
                >
                  geoseong
                </Text>
                <Text
                  style={
                    Object {
                      "borderColor": "rgb(233,237,244)",
                      "borderWidth": 0.3,
                      "color": "rgb(65,77,107)",
                      "fontSize": 12,
                      "maxWidth": 134.2,
                      "paddingHorizontal": 8,
                      "paddingVertical": 4,
                      "position": "absolute",
                      "right": 20,
                    }
                  }
                >
                  healthy
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "width": "100%",
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "borderBottomWidth": 1,
                    "borderColor": "rgb(245,245,245)",
                    "flexDirection": "row",
                    "height": 80,
                    "justifyContent": "flex-start",
                    "paddingHorizontal": 20,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "testUri": "../../../assets/icons/icon.png",
                    }
                  }
                  style={
                    Object {
                      "height": 40,
                      "width": 40,
                    }
                  }
                />
                <Text
                  style={
                    Object {
                      "color": "rgb(65,77,107)",
                      "fontSize": 14,
                      "marginLeft": 12,
                      "width": 100,
                    }
                  }
                >
                  hyochan
                </Text>
                <Text
                  style={
                    Object {
                      "borderColor": "rgb(233,237,244)",
                      "borderWidth": 0.3,
                      "color": "rgb(65,77,107)",
                      "fontSize": 12,
                      "maxWidth": 134.2,
                      "paddingHorizontal": 8,
                      "paddingVertical": 4,
                      "position": "absolute",
                      "right": 20,
                    }
                  }
                >
                  healthy
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "width": "100%",
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "borderBottomWidth": 1,
                    "borderColor": "rgb(245,245,245)",
                    "flexDirection": "row",
                    "height": 80,
                    "justifyContent": "flex-start",
                    "paddingHorizontal": 20,
                  }
                }
              >
                <Unknown
                  color="rgb(65,77,107)"
                  light={true}
                  name="meh"
                  size={40}
                />
                <Text
                  style={
                    Object {
                      "color": "rgb(65,77,107)",
                      "fontSize": 14,
                      "marginLeft": 12,
                      "width": 100,
                    }
                  }
                >
                  test
                </Text>
                <Text
                  style={
                    Object {
                      "borderColor": "rgb(233,237,244)",
                      "borderWidth": 0.3,
                      "color": "rgb(65,77,107)",
                      "fontSize": 12,
                      "maxWidth": 134.2,
                      "paddingHorizontal": 8,
                      "paddingVertical": 4,
                      "position": "absolute",
                      "right": 20,
                    }
                  }
                >
                  status
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "width": "100%",
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "borderBottomWidth": 1,
                    "borderColor": "rgb(245,245,245)",
                    "flexDirection": "row",
                    "height": 80,
                    "justifyContent": "flex-start",
                    "paddingHorizontal": 20,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "testUri": "../../../assets/icons/btnBack.png",
                    }
                  }
                  style={
                    Object {
                      "height": 40,
                      "width": 40,
                    }
                  }
                />
                <Text
                  style={
                    Object {
                      "color": "rgb(65,77,107)",
                      "fontSize": 14,
                      "marginLeft": 12,
                      "width": 100,
                    }
                  }
                >
                  geoseong
                </Text>
                <Text
                  style={
                    Object {
                      "borderColor": "rgb(233,237,244)",
                      "borderWidth": 0.3,
                      "color": "rgb(65,77,107)",
                      "fontSize": 12,
                      "maxWidth": 134.2,
                      "paddingHorizontal": 8,
                      "paddingVertical": 4,
                      "position": "absolute",
                      "right": 20,
                    }
                  }
                >
                  healthy
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "width": "100%",
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "borderBottomWidth": 1,
                    "borderColor": "rgb(245,245,245)",
                    "flexDirection": "row",
                    "height": 80,
                    "justifyContent": "flex-start",
                    "paddingHorizontal": 20,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "testUri": "../../../assets/icons/icon.png",
                    }
                  }
                  style={
                    Object {
                      "height": 40,
                      "width": 40,
                    }
                  }
                />
                <Text
                  style={
                    Object {
                      "color": "rgb(65,77,107)",
                      "fontSize": 14,
                      "marginLeft": 12,
                      "width": 100,
                    }
                  }
                >
                  hyochan
                </Text>
                <Text
                  style={
                    Object {
                      "borderColor": "rgb(233,237,244)",
                      "borderWidth": 0.3,
                      "color": "rgb(65,77,107)",
                      "fontSize": 12,
                      "maxWidth": 134.2,
                      "paddingHorizontal": 8,
                      "paddingVertical": 4,
                      "position": "absolute",
                      "right": 20,
                    }
                  }
                >
                  healthy
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
